File: pinyougou-parent/pinyougou-user-interface/src/main/java/com/pinyougou/user/service/UserService.java
package com.pinyougou.user.service;

import com.pinyougou.pojo.TbUser;

/**
 * @Author: Feng.Wang
 * @Company: Zelin.ShenZhen
 * @Description:
 * @Date: Create in 2019/5/9 10:55
 */
public interface UserService {
    public void add(TbUser user);

    /**
     * 生成验证码
     * @param phone
     */
    void createCode(String phone);

    boolean isCheckCode(String checkCode,String phone);
}

File: pinyougou-parent/pinyougou-user-web/src/main/webapp/js/service/userService.js
app.service("userService",function ($http) {
    //根据手机号得到验证码
    this.getCheckCode=function (phone) {
        return $http.get("./user/getCheckCode.do?phone="+phone);
    }
    //添加用户
    this.add=function (entity,checkCode) {
        return $http.post("./user/add.do?checkCode="+checkCode,entity);
    }
})
